mis_builder/models/mis_builder.py

```diff
@@ -0,0 +1,829 @@
+         'Period name should be unique by report'),
+    ]
+
+    def compute_domain(self, cr, uid, ids, account_, context=None):
+        if isinstance(ids, (int, long)):
+            ids = [ids]
+        domain = []
+        this = self.browse(cr, uid, ids, context=context)[0]
+        env = Environment(cr, uid, {})
+        aep = AccountingExpressionProcessor(env)
+        aep.parse_expr(account_)
+        aep.done_parsing([('company_id', '=',
+                           this.report_instance_id.company_id.id)])
+        domain.extend(aep.get_aml_domain_for_expr(account_))
+        if domain != []:
+            # compute date/period
+            period_ids = []
+            date_from = None
+            date_to = None
+            period_obj = self.pool['account.period']
+            target_move = this.report_instance_id.target_move
+            if target_move == 'posted':
+                domain.append(('move_id.state', '=', target_move))
+            if this.period_from:
+                compute_period_ids = period_obj.build_ctx_periods(
+                    cr, uid, this.period_from.id, this.period_to.id)
+                period_ids.extend(compute_period_ids)
+            else:
+                date_from = this.date_from
+                date_to = this.date_to
+            if period_ids:
+                if date_from:
+                    domain.append('|')
+                domain.append(('period_id', 'in', period_ids))
+            if date_from:
+                domain.extend([('date', '>=', date_from),
+                               ('date', '<=', date_to)])
+        else:
+            domain = False
+        return domain
+
+    def compute_period_domain(self, cr, uid, period_report, is_solde,
+                              is_initial, context=None):
+        period_obj = self.pool['account.period']
+        move_obj = self.pool['account.move']
+        domain_list = []
+        target_move = period_report.report_instance_id.target_move
+        if target_move == 'posted':
+            domain_list.append(('move_id.state', '=', target_move))
+        if not is_solde and not is_initial:
+            if period_report.period_from:
+                compute_period_ids = period_obj.build_ctx_periods(
+                    cr, uid, period_report.period_from.id,
+                    period_report.period_to.id)
+                domain_list.extend([('period_id', 'in', compute_period_ids)])
+            else:
+                domain_list.extend([('date', '>=', period_report.date_from),
+                                    ('date', '<=', period_report.date_to)])
+        elif period_report.period_from and period_report.period_to:
+            period_to = period_report.period_to
+            if is_initial:
+                move_id = move_obj.search(
+                    cr, uid, [('period_id.special', '=', False),
+                              ('period_id.date_start', '<',
+                               period_to.date_start)],
```

Comment:
If I make the search with admin in multi-company environment I get the following error :

  File "/home/lga/project/openerp/addons-account-financial-reporting/mis_builder/models/mis_builder.py", line 800, in compute
    cr, uid, lang_id, period, aep, context=context)
  File "/home/lga/project/openerp/addons-account-financial-reporting/mis_builder/models/mis_builder.py", line 618, in _compute
    context=context)
  File "/home/lga/project/openerp/addons-account-financial-reporting/mis_builder/models/mis_builder.py", line 567, in compute_period_domain
    cr, uid, computed_period_from, period_to.id)
  File "/home/lga/project/openerp/parts/openerp/addons/account/account.py", line 1065, in build_ctx_periods
    raise osv.except_osv(_('Error!'), _('You should choose the periods that belong to the same company.'))
except_osv: (u'Error!', u'You should choose the periods that belong to the same company.')